From 6531a2529d04aaa4a40dd8e312fb3c728ba632ff Mon Sep 17 00:00:00 2001
From: Takahiro Ueda <t.ueda.od@juntendo.ac.jp>
Date: Sat, 4 Nov 2023 15:42:44 +0900
Subject: [PATCH] build(doxygen): fix Doxygen failure in case of tarballs
 (#462)

- The version information file for Doxygen (version.sh.in) should be put
  into a tarball even when Doxygen is unavailable in the system where
  the tarball is created.

- version.sh is not needed for a tarball (version.sh.in is).

- The fallback in the absence of version information in configure.ac
  must also consider version.sh.in.

Close #462.
---
 configure.ac            | 1 +
 doc/doxygen/Makefile.am | 8 +++-----
 2 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 06db3a44..84c630df 100644
--- a/configure.ac
+++ b/configure.ac
@@ -25,6 +25,7 @@ m4_define([FORM_VERSION], m4_esyscmd_s([
 				echo '\\def\\repomajorversion'"{$major_version}"  >doc/manual/version.tex.in
 				echo '\\def\\repominorversion'"{$minor_version}" >>doc/manual/version.tex.in
 				cp doc/manual/version.tex.in doc/devref/version.tex.in
+				touch doc/doxygen/version.sh.in
 			fi
 			cat <<END >&2
 ========================================================================
diff --git a/doc/doxygen/Makefile.am b/doc/doxygen/Makefile.am
index adc1bbae..c40312c3 100644
--- a/doc/doxygen/Makefile.am
+++ b/doc/doxygen/Makefile.am
@@ -1,8 +1,3 @@
-######################################## CONFIG_DOXYGEN
-if CONFIG_DOXYGEN
-
-EXTRA_DIST = version.sh
-
 .PHONY: update_version
 
 # Automatic versioning.
@@ -31,6 +26,9 @@ DISTHOOK_VERSION = \
 
 endif
 
+######################################## CONFIG_DOXYGEN
+if CONFIG_DOXYGEN
+
 RUN_DOXYGEN = source ./version.sh && PROJECT_NUMBER=$$repo_version_desc ${DOXYGEN}
 
 # HTML
